save_scummer/backup.py

```diff
@@ -53,8 +54,8 @@ def make_backup(title: str, short_desc: str = None) -> str:
 
     # Determine backup path & filename
     last_save_time = get_latest_modified([path[0] for path in paths])
-    short_desc = '-' + short_desc.lower().replace(' ', '_') if short_desc else ''
-    archive_path = backup_dir.joinpath(f'{title}-{last_save_time.isoformat()}{short_desc}.zip')
+    short_desc = '-' + short_desc if short_desc else ''
+    archive_path = backup_dir / slugify(f'{title}-{last_save_time.isoformat()}{short_desc}.zip', lowercase=False)
```

Comment:
`lowercase=False` to keep the `T` in the timestamp uppercase. Note that this also drops the `.lower()` on `short_desc`; just noticed that after pushing, don't really mind either way :D